manifests/server/database.pp

```diff
@@ -34,11 +36,10 @@
     $version = $postgresql::server::_version
   }
 
-  # If the connection settings do not contain a port, then use the local server port
   if $connect_settings != undef and 'PGPORT' in $connect_settings {
-    $port = undef
+    $port_override = $port
   } else {
-    $port = $postgresql::server::port
+    $port_override = $port
   }
```

Comment:
Both branches are the same, so it could be simplified to just `$port_override = $port`.